chiware/stats.go

```diff
@@ -43,12 +47,15 @@ func StatMiddleware(ctx context.Context, router chi.Router, s stats.Stats, compo
 			defer atomic.AddInt32(&concurrentRequests, -1)
 
 			next.ServeHTTP(sw, r)
-
+			reqType := getPath(r)
+			if !chi.Routes(router).Match(chi.RouteContext(r.Context()), r.Method, reqType) {
+				reqType = UnknownPath
+			}
```

Comment:
We may simply modify `getPath` function

```go
	// getPath retrieves the path from the request.
	// The matched route's template is used if a match is found,
	// otherwise the request's URL path is used instead.
	getPath := func(r *http.Request) string {
		if path := chi.RouteContext(r.Context()).RoutePattern(); path != "" {
			return path
		}
		if conf.redactUnknownPaths {
			return "/redacted"
		}
		return r.URL.Path
	}
```
```suggestion
```

Reply:
good point, modified.